backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,211 @@
+        this.backtraceHandlerThread = backtraceHandlerThread;
+        this.api = api;
+        this.submissionUrl = settings.getBaseUrl() + "/" + urlPrefix + "/submit?token=" + settings.getToken() + "&universe=" + settings.getUniverseName();
+        this.timeBetweenRetriesMillis = settings.getTimeBetweenRetriesMillis();
+
+        long timeIntervalMillis = settings.getTimeIntervalMillis();
+
+        BacktraceAttributes backtraceAttributes = new BacktraceAttributes(context, null, null);
+        this.application = backtraceAttributes.getApplicationName();
+        this.appVersion = backtraceAttributes.getApplicationVersionOrEmpty();
+
+        if (timeIntervalMillis != 0) {
+            final BacktraceEventsHandler handler = this;
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    handler.send();
+                    handler.postDelayed(this, timeIntervalMillis);
+                }
+            }, timeIntervalMillis);
+        }
+    }
+
+    /**
+     * Number of events in the queue
+     */
+    public int getCount() {
+        return events.size();
+    }
+
+    public int getMaximumNumberOfEvents() {
+        return this.maximumNumberOfEvents;
+    }
+
+    public void setMaximumNumberOfEvents(int maximumNumberOfEvents) {
+        this.maximumNumberOfEvents = maximumNumberOfEvents;
+    }
+
+    public abstract void sendStartupEvent(String eventName);
+
+    public void send() {
+        if (events == null) {
+            return;
+        }
+        if (events.size() == 0) {
+            return;
+        }
+        sendEvents(events);
+    }
+
+    protected abstract void sendEvents(ConcurrentLinkedDeque<T> events);
+
+    protected abstract void sendEventsPayload(EventsPayload<T> payload);
+
+    @Override
+    public void handleMessage(Message msg) {
+        final BacktraceHandlerInputEvents input = (BacktraceHandlerInputEvents) msg.obj;
+        EventsResult result;
+
+        if (input.eventsRequestHandler != null) {
+            BacktraceLogger.d(LOG_TAG, "Sending using custom request handler");
+            result = input.eventsRequestHandler.onRequest(input.payload);
+        } else {
+            BacktraceLogger.d(LOG_TAG, "Sending report using default request handler");
+            String json = BacktraceSerializeHelper.toJson(input.payload);
+            result = BacktraceReportSender.sendEvents(submissionUrl, json, input.payload, input.serverErrorEventListener);
+        }
+
+        if (input.eventsOnServerResponseEventListener != null) {
+            BacktraceLogger.d(LOG_TAG, "Processing result using custom event");
+            input.eventsOnServerResponseEventListener.onEvent(result);
+        }
+
+        int statusCode = result.getStatusCode();
+        if (statusCode > HttpURLConnection.HTTP_NOT_IMPLEMENTED && statusCode != HttpURLConnection.HTTP_VERSION) {
+            int numRetries = ++input.payload.numRetries;
```

Comment:
extract body of this if statement to method retrySendEvents(); or with similar name

Reply:
Will do!